R/rwkv_cpp/build_tarballs.jl

```diff
@@ -0,0 +1,67 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "rwkv_cpp"
+version = v"0.0.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/RWKV/rwkv.cpp.git", "e8b6adf75283b467cdb5739e55324cf6d68020d0"),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/rwkv.cpp
+
+for f in ${WORKSPACE}/srcdir/patches/*.patch; do
+    atomic_patch -p1 ${f}
+done
+
+git submodule init
+git submodule update
+
+cmake -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -G Ninja \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DRWKV_AVX=OFF \
+    -DRWKV_AVX2=OFF \
+    -DRWKV_STANDALONE=ON
+
+cmake --build build --parallel ${nproc}
+cmake --install build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "freebsd"; ),
+    Platform("i686", "windows"; ),
+    Platform("x86_64", "windows"; )
+]
```

Comment:
What's the problem with the other platforms?

Reply:
For `i686`, `armv6l`, `armv7l`

```
[17:54:51] /workspace/srcdir/rwkv.cpp/rwkv.cpp:37:1: error: static assertion failed: File offsets should be 64-bit or else rwkv.cpp will not be able to load model files over 2 GB
```

For `aarch64`

```
[17:55:58] In file included from /workspace/srcdir/rwkv.cpp/ggml/src/ggml.c:3:0:
[17:55:58] /workspace/srcdir/rwkv.cpp/ggml/include/ggml/ggml.h:283:13: error: unknown type name ‘__fp16’
[17:55:58]      typedef __fp16 ggml_fp16_t;
[17:55:58]              ^
```

Reply:
You say `i686` has error but it is actually in the list of platforms?

What about macOS and freebsd, why are they missing?

In general I would recommend to start with the full list of platforms, then filtering it down. So for example:
```
platforms = supported_platforms()
platforms = filter!(p -> arch(p) != "armv6l", platforms)
platforms = filter!(p -> arch(p) != "armv7l", platforms)
```
(It may also be possible to fix the first error by adding `-D_FILE_OFFSET_BITS=64` to `CPPFLAGS`).